scripts/hcl-check.sh

```diff
@@ -0,0 +1,312 @@
+    echo ""
+    exit 0
+}
+
+# extract_hcl searches for .markdown files by using glob 'website/docs/{*/,?}*markdown' in:
+# * website/docs/*.markdown
+# * website/docs/*/*.markdown
+# It will look for code blocks starting with '```hcl' and extract their contents until closing '```'
+# and store in a file inside a tmp directory. Filename will be "base_filename+total_occurence_number"
+# (e.g. edgegateway.html.markdown-100.tf)
+function extract_hcl {
+    curdir=$PWD
+
+    [ -d "$tmp_dir" ] && rm -r "$tmp_dir"
+    mkdir "$tmp_dir"
+    cd "$tmp_dir" || exit 1
+
+    awk 'function basename(file) {
+        sub(".*/", "", file)
+        return file
+    } /^```hcl/ {
+    flag = 1
+    ++n
+    s = ""
+    next
+    }
+    /^```$/ {
+        if (flag==1) {
+            print_verbose "Extracting # ${n} HCL block in file ${basename(FILENAME)}"
+            print s > (basename(FILENAME) "-" n ".tf"); close((basename(FILENAME) "-" n ".tf"))
+        }
+        flag = 0
+    }
+    flag {
+    s = s $0 ORS
+    }' ../website/docs/{*/,?}*markdown
+    hcl_number="$(ls | wc -l | tr -d ' ')"
+    cd "$curdir" || exit 1
+}
+
+# Runs the fmt subcommand of Terraform to check syntax errors on temporary HCL files
+# that contain the documentation snippets.
+function terraform_fmt_check {
+    fmt_error_text=''
+
+    for hcl_file in "$tmp_dir"/*.tf
+    do
+        command_start_time=$(date +%s)
+        print_progress "Checking format of $hcl_file..."
+        terraform fmt -check $hcl_file &>/dev/null
+        retVal=$?
+        if [ $retVal -ne 0 ]; then
+            ((fmt_errors++))
+            fmt_error_text="${fmt_error_text}$(terraform fmt -no-color -diff -check "$hcl_file" 2>&1)"
+        fi
+        command_end_time=$(date +%s)
+        print_times "terraform fmt $hcl_file" "$command_start_time" "$command_end_time"
+    done
+
+    echo ""
+}
+
+# Runs against all generate HCL files and checks if there are no two (or more) empty lines
+function multi_newline_check {
+  newline_error_text=''
+
+  for hcl_file in "$tmp_dir"/*.tf
+      do
+        print_progress "Checking for multiple newlines in $hcl_file..."
+
+        hcl_content=$(cat $hcl_file)
+        # 'cat -s' squeezes multiple adjacent empty lines, causing the output to be single spaced.
+        hcl_content_without_newlines=$(cat -s $hcl_file)
+
+        if [ "$hcl_content_without_newlines" != "$hcl_content" ]; then
+          newline_error_text="${newline_error_text} \n ${hcl_file}"
+          ((newline_errors++))
+        fi
+      done
+
+      echo ""
+}
+
+# Performs a Terraform init on a temporary HCL file that contain the documentation snippets.
+# If the init command fails, gives an error message and the script will fail.
+function terraform_validation_check {
+    init_error_text=''
+    provider_version="$(cat VERSION)"
+    git tag "$provider_version" > /dev/null 2>&1
+    if [ "$?" -eq 0 ]; then
+      clean_tag=1
+    fi
+    make install > /dev/null 2>&1
```

Comment:
This part diverges from the original, as it did `git describe` but we don't have tags.
Now, it just installs what is currently on HEAD.